### PR TITLE
BotLog model validation

### PR DESCRIPTION
**Pull request by dtoledo23**
#### What this PR does?
- Defines a model for the incomming logs
- Add validation layer to avoid enqueueing bad logs into the analytics stream
- Documentation

#### Where should the reviewer start?
Put special atention reviewing the schema. We need to make sure it is alright and it is what we are expecting.

#### How should this be manually tested?
- Set your AWS credentials as environment variables. [Here](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-environment)
- The Spark application and Postgress should be running already. Make sure you have access to Postgress.
- Run the example: `node examples/single-log-example.js`
- It should not throw any error and after some second there should be results in Postgress
- Mess around with the object being logged only to confirm that the schema works and it throws an error if the object does not follow the schema defined.

#### Any background context you want to provide?
This is going to be used by the backend of any bot we are going to analyze.

#### What are the relevant tickets?
- [TB-17](https://wizeline.atlassian.net/browse/TB-17)
- [TB-43](https://wizeline.atlassian.net/browse/TB-43)

- [x] @juanitodread

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # wizeline-bot-broker-logger
 
-wizeline-bot-broker-logger
+Utility module for logging bot events into the analytics platforms.
```
Comment (juanitodread) [inline README.md]: I think "platforms" should be updated to singular => "platform"

File: README.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 ## How to use
 
-* Install dependancy:
+### Install dependancy
```
Comment (juanitodread) [inline README.md]: dependency

File: README.md [hunk 3]
```diff
@@ -24,13 +24,44 @@ let logger = new Logger({
...
+    "stage" : "qerg",
+    "timestamp" : Date.now(),
+    "conversationType" : "TWG",
+    "conversationState" : "started",
+}
+
+logger.analytics(logObject);
+```
+
+> See the complete example under `/examples`
```
Comment (juanitodread) [inline README.md]: Why not put "/examples" as a link?

File: README.md [hunk 4]
```diff
@@ -24,13 +24,44 @@ let logger = new Logger({
...
+- `botId`
+- `sessionId`
+- `stage`
+- `timestamp`
+- `conversationType`
+- `conversationState`
+
+The analytics platform needs also the flags:
+
+- Stage: To define the stage of the bot backend processing system on which this event hapenned
```
Comment (juanitodread) [inline README.md]: happened

File: models/BotLog/BotLog.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        const validatorResult = validate(obj, BotLogSchema);
+
+        if (!validatorResult.valid) {
+          throw new Error("Invalid BotLog object. " + validatorResult.errors);
+        }
+    }
+
+
+    toString() {
+        return JSON.stringify(this.obj);
```
Comment (juanitodread) [inline models/BotLog/BotLog.js]: Very good abstraction! 👍

Comment (dtoledo23) [inline models/BotLog/BotLog.js]: 🙊

**Pull request merged**